lib/icingadb/icingadb-objects.cpp

```diff
@@ -2589,6 +2593,16 @@ void IcingaDB::StateChangeHandler(const ConfigObject::Ptr& object, const CheckRe
 	}
 }
 
+void IcingaDB::ReachabilityChangeHandler(const std::set<Checkable::Ptr>& children)
+{
+	for (const IcingaDB::Ptr& rw : ConfigType::GetObjectsByType<IcingaDB>()) {
+		for (auto& checkable : children) {
+			rw->UpdateState(checkable);
+			rw->SendStatusUpdate(checkable);
```

Comment:
Heads-up notice: this change conflicts with #9144 so it might make sense to first review that PR, hopefully merge it and then update this PR.